modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/RootQuery.java

```diff
@@ -65,10 +67,10 @@
     /** Parameters. */
     private final Object[] params;
 
-    /** remote nodes */
-    private final Set<UUID> remotes;
+    /** Remote nodes unfinished fragments count. */
+    private final Map<UUID, AtomicInteger> remoteFragments;
```

Comment:
I suppose that simple Integer would be enough here, possible conc part defended be mux monitor, i recheck with Integer and all tests is ok, did i miss smth ?

Reply:
Yes, a simple Integer is enough, but Integer is a boxed type, a new instance of Integer should be created on each increment. `AtomicInteger` here used as int holder. It can be also something like `int[]`, but as for me, atomic integer is more readable.

Reply:
oh, seems we need a comment here, cause someone want`s to fix it )

Reply:
I've added the comment.